project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module

```diff
@@ -786,3 +817,61 @@ function c4m_content_taxonomy_build_tree($items, $link_term_to_facet = array())
 
   return theme_html_tag($tag);
 }
+
+/**
+ * Helper function.
+ *
+ * Extracts all child taxonomy items from $items array (alters it),
+ * and returns them.
+ *
+ * @param array $items
+ *   Taxonomies array, from which we extract all children of item with
+ *   ID = $parent_id. $items array is altered by function.
+ * @param int $parent_id
+ *   The ID of parent taxonomy item.
+ *
+ * @return array $result
+ *   All child items of input parent ID.
+ */
+function _c4m_content_taxonomy_extract_child_items(&$items, $parent_id) {
+  $result = array();
+  $remaining = array();
+
+  foreach ($items as $item) {
+    $parent = taxonomy_get_parents($item['tid']);
+    if (!empty($parent) &&  key($parent) == $parent_id) {
```

Comment:
`if (!empty($parent) && key($parent) == $parent_id) {` (extra space)

plus, add a comment about the logic of the condition.